modules/extensions-xml-support/src/main/java/org/mule/runtime/extension/internal/ast/MacroExpandedComponentAst.java

```diff
@@ -107,14 +112,20 @@ public <T> Either<String, T> getValue() {
         }
 
         return (Either<String, T>) originalValue
-            .mapRight(v -> {
-              if (v instanceof String) {
-                return macroExpandedRawValue((String) v);
-              } else if (getDecorated().getModel().getName().equals(ERROR_MAPPINGS_PARAMETER_NAME)) {
-                return mapErrorMappings(getDecorated());
-              }
-              return v;
-            });
+            .mapRight(this::mapComponent);
+      }
+
+      private <T> Object mapComponent(T rawValue) {
+        if (rawValue instanceof ArrayList) {
+          return rawValue;
+        } else if (rawValue instanceof String) {
+          return macroExpandedRawValue((String) rawValue);
```

Comment:
Add a comment that can be something like this: "This is done to evaluate if the parameter is not declared as a child and can be an expression, so that it is resolved"

Reply:
Done. Please, review it.